### DISCUSSION TITLE
autorender a list of elements by a custom CSS selector / custom HTML tag type

### DISCUSSION THREAD

**Discussion by vadimkantorov** [KaTeX/KaTeX#3946] [General]
Hi!

Is it possible to pass a custom CSS selector / array of HTML elements produced by `document.querySelectorAll('code.my-math-block')` or `document.querySelectorAll('code.my-math-inline')` to `renderMathInElement(...)` ?

Currently even `<code class="my-math-block">$$2+\frac{1}{x}$$</code>` does not get rendered (only works if I change the tag name from `code` to `span`)

Or is there an alternative API to achieve this nicely? (i.e. I have `code` tags with some classes for inline-style and block-style math expressions)

Should I use some hacks like:
`<script defer src="https://cdn.jsdelivr.net/npm/katex@0.16.10/dist/katex.min.js" integrity="sha384-hIoBPJpTUs74ddyc4bFZSM1TVlQDA60VBbJS0oA934VSz82sBx1X7kSx2ATBDIyd" crossorigin="anonymous" onload="document.querySelectorAll('code.my-math-block').forEach(elem=>katex.render(elem.innerText,elem,{displayMode:true,throwOnError:false}));"></script>` to render `<code class="my-math-block">2+\frac{1}{x}</code>` somewhere in my HTML?

**Comment by edemaine** [comment]
Yes, your hack sounds like what you want. Some similar code is in <https://github.com/KaTeX/KaTeX/tree/main/contrib/mathtex-script-type>, which is quite short, so could be the basis for what you want. I could see adding an API for this, though I'm not sure exactly where... Perhaps this contrib module or a new one.

**Reply by vadimkantorov** [reply]
I was also not sure if KaTeX allows to do rendering of many elements somehow in some worker thread pool. I was fearing that a plain `forEach` would not trigger the parallel/off-UI-thread rendering. Hoping that KaTeX has smarts for this to avoid blocking of the UI by doing heavy-rendering :)

Too bad we cannot legally abuse `<math display="inline">arbitrary expression</math>` or `<math display="block">arbitrary expression</math>` :)
